homeassistant/components/tplink/switch.py

```diff
@@ -121,8 +126,7 @@ def _plug_from_context(self):
         children = self.smartplug.sys_info["children"]
         return next(c for c in children if c["id"] == self.smartplug.context)
 
-    def update(self):
-        """Update the TP-Link switch's state."""
+    async def async_attempt_update(self):
```

Comment:
If the function does sync I/O it can't be a coroutine.

Reply:
Changed back to sync function.